### PR TITLE
Add ability to set/replace the prompt

### PR DESCRIPTION
**Pull request by KidVizious**
This PR adds the ability for the user to set the prompt.  If the prompt is not set or the user un-sets it, the default prompt will be used instead.

### CODE REVIEW

File: src/ecdc/ecdc.c [hunk 1]
```diff
@@ -825,6 +835,26 @@ ecdc_alloc_list_command(struct ecdc_console * console,
...
+char const * ecdc_replace_prompt(struct ecdc_console *console, char const *prompt)
+{
+    if ((NULL == console) || (NULL == prompt))
+    {
+        return NULL;
+    }
+
+    // If the prompt was previously not set, it will be NULL which will not
+    // have any ill side effects when calling free
+    free(console->prompt);
```
Comment (bradschl) [inline src/ecdc/ecdc.c:850]: Do this in ecdc_console_free(), (~line 750) as well

File: src/ecdc/ecdc.c [hunk 2]
```diff
@@ -825,6 +835,26 @@ ecdc_alloc_list_command(struct ecdc_console * console,
 }
 
 
+char const * ecdc_replace_prompt(struct ecdc_console *console, char const *prompt)
```
Comment (bradschl) [inline src/ecdc/ecdc.c]: Update format to be consistent with the rest of the file
```
char const *
ecdc_replace_prompt(struct ecdc_console *console, char const *prompt)
{
    // ....
}
```

File: src/ecdc/ecdc.c [hunk 3]
```diff
@@ -825,6 +835,26 @@ ecdc_alloc_list_command(struct ecdc_console * console,
...
+
+    // If the prompt was previously not set, it will be NULL which will not
+    // have any ill side effects when calling free
+    free(console->prompt);
+
+    console->prompt = strdup(prompt);
+    return console->prompt;
+}
+
+void ecdc_free_prompt(struct ecdc_console *console)
```
Comment (bradschl) [inline src/ecdc/ecdc.c]: Update format to be consistent with the rest of the file
```
void
ecdc_free_prompt(struct ecdc_console *console)
{
    // ...
}
```

File: src/ecdc/ecdc.c (hunk 1 above)
Comment (KidVizious) [reply src/ecdc/ecdc.c:850]: Done.

File: src/ecdc/ecdc.c (hunk 2 above)
Comment (KidVizious) [reply src/ecdc/ecdc.c]: Done.

File: src/ecdc/ecdc.c (hunk 3 above)
Comment (KidVizious) [reply src/ecdc/ecdc.c]: Done.

File: src/ecdc/ecdc.h [hunk 4]
```diff
@@ -217,6 +217,31 @@ struct ecdc_command *
...
+
+/**
+ * @brief Deallocated previously allocated prompt
+ * @details This will deallocate any resources allocated by the ecdc_set_prompt command
+ *          and resets prompt to default.
+ * 
+ * @param   ecdc_console Console for which to deallocate prompt
+ */
+void 
+ecdc_free_prompt(struct ecdc_console *console);
```
Comment (bradschl) [inline src/ecdc/ecdc.h]: Please move these signatures up to line ~149

**Comment by bradschl** [comment]
Released as package 0.0.3

**Pull request merged**